pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -180,6 +219,113 @@ func createRawWorkerDeployment(componentMeta metav1.ObjectMeta,
 	return deployment
 }
 
+func GetKServeContainerPort(podSpec *corev1.PodSpec) string {
+	for _, container := range podSpec.Containers {
+		if container.Name == "kserve-container" {
+			if len(container.Ports) > 0 {
+				return strconv.Itoa(int(container.Ports[0].ContainerPort))
```

Comment:
Most of the port types on K8s are https://github.com/kubernetes/apimachinery/blob/master/pkg/util/intstr/intstr.go.
imho it is better to use it and avoid casts.

Reply:
we need the string here because the port value is used as a arg for the oauth proxy container